include/miral/miral/idle_listener.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIRAL_IDLE_LISTENER_H
+#define MIRAL_IDLE_LISTENER_H
+
+#include <functional>
+#include <memory>
+
+namespace mir { class Server; }
+
+namespace miral
+{
+
+struct IdleListenerParams
```

Comment:
I don't think "Params" is good naming. And if it were, then `IdleListener::Params` would probably be better scoping. Maybe `IdleListener::Callbacks`?

Reply:
It does sound like there were more `Params` before, now they're just callbacks?

Also `IdleListener::Callbacks::Callback` would probably need moving up?

Reply:
Updated to:
- `IdleListener::Callbacks`
- `IdleListener::Callback`